### PR TITLE
TOML docs: s2s

### PR DESCRIPTION
**Pull request by chrzaszcz**
Document the s2s options.

See the commit messages for details.

### CODE REVIEW

**Review by NelsonVides — COMMENTED** [review]
I'm only thinking now that I realise, that some sections, like the previous acl and access ones, mark the options with three heading levels (`###`), others like the listen section have four such levels (`####`), but here s2s options have two (`##`). Shouldn't we make them all consistent? Could be a topic for a separate styling PR, maybe?

I was also surprised to see this new `host_policy` option, couldn't find it in the old docs, nor how does it work in the code, could you point me out how to find it? 🤔

**Comment by chrzaszcz** [comment]
> I'm only thinking now that I realise, that some sections, like the previous acl and access ones, mark the options with three heading levels (`###`), others like the listen section have four such levels (`####`), but here s2s options have two (`##`). 
The levels descend from the top - so if there are no subsections, it would be '##' and with one level of subsections it would be `###`. This rule should be applied consistently.

> Shouldn't we make them all consistent? Could be a topic for a separate styling PR, maybe?
The only problem I see is the readability - the `readthedocs` theme makes the `code`- formatted headers less pronounced when the font is smaller. IMO this is outside of the scope and maybe the solution would be to change the theme as @gustawlippa once suggested. 

> I was also surprised to see this new `host_policy` option, couldn't find it in the old docs, nor how does it work in the code, could you point me out how to find it? 🤔
It was called `s2s_host` before. The name was less descriptive.

File: doc/advanced-configuration/s2s.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+Defines a list of accepted SSL ciphers for outgoing S2S connections.
+Please refer to the [OpenSSL documentation](http://www.openssl.org/docs/apps/ciphers.html) for the cipher string format.
+
+## `s2s.max_retry_delay`
+* **Scope:** local
+* **Syntax:** positive integer
+* **Default:** `300`
+* **Example:** `max_retry_delay = 300`
+
+Specifies the maximum time in seconds that MongooseIM will wait until next attempt to connect to a remote XMPP server. The delays between consecutive attempts will be doubled until this limit is reached.
```
[the remark this answers is not in the document]
Comment (kmakiela) [reply doc/advanced-configuration/s2s.md]: > I'm only thinking now that I realise, that some sections, like the previous acl and access ones, mark the options with three heading levels (`###`), others like the listen section have four such levels (`####`), but here s2s options have two (`##`). Shouldn't we make them all consistent? Could be a topic for a separate styling PR, maybe?
> 
> I was also surprised to see this new `host_policy` option, couldn't find it in the old docs, nor how does it work in the code, could you point me out how to find it? 🤔
This option was named `s2s_host` before and it was described in the [Advanced Configuration](https://github.com/esl/MongooseIM/blob/master/doc/Advanced-configuration.md)

**Review by kmakiela — APPROVED** [review]
Thanks for catching the missing `s2s_host` option

**Comment by NelsonVides** [comment]
> > I'm only thinking now that I realise, that some sections, like the previous acl and access ones, mark the options with three heading levels (`###`), others like the listen section have four such levels (`####`), but here s2s options have two (`##`).
> 
> The levels descend from the top - so if there are no subsections, it would be '##' and with one level of subsections it would be `###`. This rule should be applied consistently.
> 
> > Shouldn't we make them all consistent? Could be a topic for a separate styling PR, maybe?
> 
> The only problem I see is the readability - the `readthedocs` theme makes the `code`- formatted headers less pronounced when the font is smaller. IMO this is outside of the scope and maybe the solution would be to change the theme as @gustawlippa once suggested.
> 
> > I was also surprised to see this new `host_policy` option, couldn't find it in the old docs, nor how does it work in the code, could you point me out how to find it? 🤔
> 
> It was called `s2s_host` before. The name was less descriptive.

Right, all makes sense. We'll be back to `readthedocs` and themes and formatting matters some other time in the future, it's out of the scope.

`host_policy` opt, also clear now 👍

**Review by NelsonVides — APPROVED** [review]

**Pull request merged**